src/main/java/bio/terra/service/dataset/DatasetService.java

```diff
@@ -316,9 +322,23 @@ public DatasetSummaryModel patch(
     if (!patchSucceeded) {
       throw new RuntimeException("Dataset was not updated");
     }
+    // TODO - what should we do if they're turning off secure monitoring?
```

Comment:
Idea -- maybe we should only support turning secure monitoring on?  I know that would make the patch DAO method a bit more complex.

Reply:
I think that'd be reasonable, especially for this quick turnaround.

Reply:
The new approach only supports enabling secure monitoring.